solr/solrj/src/java/org/apache/solr/common/cloud/ZkNodeProps.java

```diff
@@ -171,8 +167,10 @@ public boolean getBool(String key, boolean b) {
   public boolean equals(Object that) {
     return that instanceof ZkNodeProps && ((ZkNodeProps)that).propMap.equals(this.propMap);
   }
-//  @Override
-//  public int hashCode() {
-//    throw new UnsupportedOperationException("TODO unimplemented ZkNodeProps.hashCode");
-//  }
+
+  //TODO: I'm very uncertain about this
+  @Override
+  public int hashCode() {
+    return 0;
```

Comment:
A constant `hashCode()` return value is correct but does not _spread_ objects into buckets in any useful way. Here we could return `propMap.hashCode()` instead.

Reply:
Yeah, it was late that night ;). Hashed propMap.